core/src/main/java/zipkin/finagle/ReporterMetricsAdapter.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.finagle;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import com.twitter.finagle.stats.Counter;
+import com.twitter.finagle.stats.StatsReceiver;
+import scala.collection.JavaConverters;
+import scala.runtime.AbstractFunction0;
+import zipkin.reporter.ReporterMetrics;
+
+import static java.util.Arrays.asList;
+
+final class ReporterMetricsAdapter implements ReporterMetrics {
+  private final Counter spans;
+  private final Counter spansDropped;
+  private final Counter spanBytes;
+  private final Counter messages;
+  private final Counter messageBytes;
+  private final StatsReceiver messagesDropped;
+  private final AtomicDouble spanQueueSize;
+  private final AtomicDouble spanQueueBytes;
+
+  public ReporterMetricsAdapter(StatsReceiver stats) {
+    this.spans = stats.counter0("spans");
+    this.spanBytes = stats.counter0("span_bytes");
+    this.spansDropped = stats.counter0("spans_dropped");
+    this.messages = stats.counter0("messages");
+    this.messageBytes = stats.counter0("message_bytes");
+    this.messagesDropped = stats.scope("messages_dropped");
+    this.spanQueueSize = gaugeFor(stats, "span_queue_size");
+    this.spanQueueBytes = gaugeFor(stats, "span_queue_bytes");
+  }
+
+  @Override public void incrementMessages() {
+    messages.incr();
+  }
+
+  @Override public void incrementMessagesDropped(Throwable cause) {
+    messagesDropped.counter0(cause.getClass().getName()).incr();
+  }
+
+  @Override public void incrementSpans(int i) {
+    spans.incr(i);
+  }
+
+  @Override public void incrementSpanBytes(int i) {
+    spanBytes.incr(i);
+  }
+
+  @Override public void incrementMessageBytes(int i) {
+    messageBytes.incr(i);
+  }
+
+  @Override public void incrementSpansDropped(int i) {
+    spansDropped.incr(i);
+  }
+
+  @Override public void updateQueuedSpans(int i) {
+    spanQueueSize.set(i);
+  }
+
+  @Override public void updateQueuedBytes(int i) {
+    spanQueueBytes.set(i);
+  }
+
+  private static AtomicDouble gaugeFor(StatsReceiver stats, String scope) {
+    final AtomicDouble result = new AtomicDouble();
+    stats.provideGauge(JavaConverters.asScalaBufferConverter(asList(scope)).asScala(),
```

Comment:
instead of converting to scala, we can use the [java-flavor API][0] here.  This instead becomes:

```java
StatsReceiver.provideGuage(stats, new Callable[Float] { public call() { return result.floatValue(); } }, asList(scope).toArray);
```

[0]: https://github.com/twitter/util/blob/develop/util-stats/src/main/scala/com/twitter/finagle/stats/StatsReceiver.scala#L31-L33